test/test_dirichlet_dist.cpp

```diff
@@ -54,183 +51,293 @@ void test_spot(
 
   // Some exact values of pdf.
-  alpha[0] = 1.0, alpha[1] = 1.0;
-  x[0] = 0.5, x[1] = 0.5;
+  alpha[0] = static_cast<RealType>(1.0), alpha[1] = static_cast<RealType>(1.0);
+  x[0] = static_cast<RealType>(0.5), x[1] = static_cast<RealType>(0.5);
   BOOST_CHECK_EQUAL(boost::math::pdf(dirichlet_distribution<V>(std::move(alpha)), x), static_cast<RealType>(1.0));
 
-  alpha[0] = 2.0, alpha[1] = 2.0;
-  x[0] = 0.5, x[1] = 0.5;
+  alpha[0] = static_cast<RealType>(2.0), alpha[1] = static_cast<RealType>(2.0);
+  x[0] = static_cast<RealType>(0.5), x[1] = static_cast<RealType>(0.5);
   BOOST_CHECK_EQUAL(boost::math::pdf(dirichlet_distribution<V>(std::move(alpha)), x), static_cast<RealType>(1.5));
 
   // Checking precalculated values on scipy.
   // https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.dirichlet.html
-  alpha[0] = static_cast<RealType>(5.778238829L); alpha[1] = static_cast<RealType>(2.55821892973L);
-  x[0] = static_cast<RealType>(0.23667289213L); x[1] = static_cast<RealType>(1.0L - 0.23667289213L);
-  V mean = {static_cast<RealType>(0.69312878L), static_cast<RealType>(0.30687122L)};
-  V mode = {static_cast<RealType>(0.75408675L), static_cast<RealType>(0.24591325L)};
-  V var = {static_cast<RealType>(0.0227818L), static_cast<RealType>(0.0227818L)};
-  V skewness = {static_cast<RealType>(-0.49515513L), static_cast<RealType>(0.49515513L)};
-  V kurtosis = {static_cast<RealType>(-139231.64518864L), static_cast<RealType>(-6993.41057616L)};
-  RealType entropy = static_cast<RealType>(17.6747L);
-  RealType pdf = static_cast<RealType>(0.05866154L);
-  RealType cdf = static_cast<RealType>(0.00071693L);
-  tolerance *= 1E+07;
+  alpha[0] = static_cast<RealType>(5.778238829);
+  alpha[1] = static_cast<RealType>(2.55821892973);
+  x[0] = static_cast<RealType>(0.23667289213);
+  x[1] = static_cast<RealType>(0.76332710787);
+  V mean = {static_cast<RealType>(0.693128783978901), static_cast<RealType>(0.3068712160210989)};
+  V var = {static_cast<RealType>(0.022781795654775592), static_cast<RealType>(0.022781795654775592)};
+  RealType entropy = static_cast<RealType>(-0.516646371355904);
+  RealType pdf = static_cast<RealType>(0.05866153821852176);
+
+  test_spot(std::move(alpha),
+            std::move(x),
+            std::move(mean),
+            std::move(var),
+            entropy, pdf, tolerance);
+
+  alpha[0] = static_cast<RealType>(5.310948003052013);
+  alpha[1] = static_cast<RealType>(8.003963132298916);
+  x[0] = static_cast<RealType>(0.35042614416132284);
+  x[1] = static_cast<RealType>(0.64957385583867716);
+  mean[0] = static_cast<RealType>(0.398872207937724);
+  mean[1] = static_cast<RealType>(0.601127792062276);
+  var[0] = static_cast<RealType>(0.016749888798155716);
+  var[1] = static_cast<RealType>(0.016749888798155716);
+  pdf = static_cast<RealType>(2.870121181949622);
+  entropy = static_cast<RealType>(-0.6347509574442718);
```

Comment:
I couldn't get higher precision than this (float64 in NumPy), because gamma function in SciPy doesn't support float128.